cachi2/core/package_managers/pip.py

```diff
@@ -1765,47 +1795,45 @@ def _process_package_distributions(
             else:
                 log.info("Filtering out %s due to checksum mismatch", wheel.filename)
 
-        if len(filtered) > 0 and len(wheels) == 0:
+        if len(filtered) == 0 and len(sdists) == 0:
+            raise PackageRejected(
+                f"No archives found for package {name}=={version}",
```

Comment:
Why are we checking the `filtered` list, rather than the `wheels` list? Isn't it possible that `filtered` is non-empty and `wheels` is empty? We should fail if we don't have any archives for a dependency

Reply:
> We should fail if we don't have any archives for a dependency

And keep in mind that "don't have any archives" can also happen after we download the wheels and delete the ones that don't hash-check